contracts/admin/src/common/type_defs.rs

```diff
@@ -0,0 +1,12 @@
+use super::{CodeIdWithMigrateMsg, GeneralContracts, LpnContracts};
+
+pub type MigrateContract = CodeIdWithMigrateMsg<String>;
+pub type MaybeMigrateContract = Option<MigrateContract>;
```

Comment:
[overingeneered] while admitting that defining separate types is a good practice, and I personally am a strong believer in that, that makes sense when having different entities. In this case, we have got three copies of the same types `MigrationSpec` and `MaybeMigrationSpec`. Please get rid of these 6 extra types.